atomisticparsers/h5md/parser.py

```diff
@@ -0,0 +1,690 @@
+
+class H5MDParser(MDParser):
+    def __init__(self):
+        super().__init__()
+        self._data_parser = HDF5Parser()
+        self._n_frames = None
+        self._n_atoms = None
+        self._atom_parameters = None
+        self._system_info = None
+        self._observable_info = None
+        self._parameter_info = None
+        self._time_unit = ureg.picosecond
+        self._path_group_particles_all = None
+        self._path_group_positions_all = None
+        self._path_value_positions_all = None
+
+        self._nomad_to_particles_group_map = {
+            'positions': 'position',
+            'velocities': 'velocity',
+            'forces': 'force',
+            'labels': 'species_label',
+            'label': 'force_field_label',
+            'mass': 'mass',
+            'charge': 'charge',
+        }
+
+        self._nomad_to_box_group_map = {
+            'lattice_vectors': 'edges',
+            'periodic': 'boundary',
+            'dimension': 'dimension'
+        }
+
+    def format_times(self, times):
+        return np.around(times.magnitude * ureg.convert(1.0, times.units, self._time_unit), 5)
+
+    def parse_atom_parameters(self):
+        if self._n_atoms is None:
+            return {}
+        self._atom_parameters = {}
+        n_atoms = self._n_atoms[0]  # TODO Extend to non-static n_atoms
+
+        atom_parameter_keys = ['label', 'mass', 'charge']
+        for key in atom_parameter_keys:
+            value = self._data_parser.get(f'{self._path_group_particles_all}.{self._nomad_to_particles_group_map[key]}')
+            if value is not None:
+                self._atom_parameters[key] = value
+            else:
+                continue
+            if isinstance(self._atom_parameters[key], h5py.Group):
+                self.logger.warning('Time-dependent atom parameters currently not supported. Atom parameter values will not be stored.')
+                continue
+            elif len(self._atom_parameters[key]) != n_atoms:
+                self.logger.warning('Inconsistent length of some atom parameters. Atom parameter values will not be stored.')
+                continue
+
+    def parse_system_info(self):
+        self._system_info = {'system': {}, 'calculation': {}}
+        particles_group = self._data_parser.get(self._path_group_particles_all)
+        positions = self._data_parser.get(self._path_value_positions_all)
+        n_frames = self._n_frames
+        if particles_group is None or positions is None or positions is None: # For now we require that positions are present in the H5MD file to store other particle attributes
+            self.logger.warning('No positions available in H5MD file. Other particle attributes will not be stored')
+            return self._system_info
+
+        def get_value(value, steps, path=''):
+            if value is None:
+                return value
+            if isinstance(value, h5py.Group):
+                path_value = f'{path}.value' if path else 'value'
```

Comment:
for cases like this, i would simply not define path_value and put the expression directly in .get(...)

Reply:
got it. done.